src/probes.cpp

```diff
@@ -173,9 +195,19 @@ void special_entry(dyntracer_t* dyntracer,
 
     Call* function_call = state.create_call(call, op, args, rho);
 
-    std::string function_name = function_call->get_function_name();
-    if (function_name.compare("<<-") == 0) {
-        state.process_dynamic_calls_for_specials(function_call);
+    Function * function = function_call->get_function();
```

Comment:
These lines are repeated, there is no need to have multiple if-else definitions.